### PR TITLE
fix: email not required for update call

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Email was marked as a required field for creating and updating lead calls, which is not the case. It is only required on creating one.
This PR fixes that along with test cases.
## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/persistiq/data/persistIqIdentifyConfig.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 [
     {
         "destKey": "email",
-        "sourceKeys": "email",
-        "required": true,
-        "sourceFromGenericMap": true
+        "sourceKeys": [
```
Comment (yashasvibajpai) [inline v0/destinations/persistiq/data/persistIqIdentifyConfig.json:4]: Shouldn't we support passing email in root also?

File: v0/destinations/persistiq/data/persistIqIdentifyConfig.json [hunk 2]
```diff
@@ -97,7 +98,10 @@
     },
     {
         "destKey": "company_name",
-        "sourceKeys": ["conapny.name", "company"],
+        "sourceKeys": [
+            "conapny.name",
```
Comment (yashasvibajpai) [inline v0/destinations/persistiq/data/persistIqIdentifyConfig.json:102]:
```suggestion
            "company.name",
```
Typo

File: __tests__/data/persistiq.json [hunk 3]
```diff
@@ -1,13 +1,12 @@
 [
   {
-    "description": "[::IDENTIFY::]-> Update call with leadId in externalId",
+    "description": "[::IDENTIFY::]-> Update call with leadId in externalId and no email",
```
Comment (yashasvibajpai) [inline __tests__/data/persistiq.json:3]: Can we add the company field in the test case too?

**Review by yashasvibajpai — CHANGES_REQUESTED** [review]

File: __tests__/data/persistiq.json (hunk 3 above)
Comment (anantjain45823) [reply __tests__/data/persistiq.json:3]: yeah sure

File: v0/destinations/persistiq/data/persistIqIdentifyConfig.json (hunk 1 above)
Comment (anantjain45823) [reply v0/destinations/persistiq/data/persistIqIdentifyConfig.json:4]: like message.email ?

Comment (yashasvibajpai) [reply v0/destinations/persistiq/data/persistIqIdentifyConfig.json:4]: Yes, as I see earlier you added the support

Comment (anantjain45823) [reply v0/destinations/persistiq/data/persistIqIdentifyConfig.json:4]: it was from generic map. But there is a problem in generic mapping for email i.e. it takes the value of 0th index of externalID as last fallback value.

Comment (anantjain45823) [reply v0/destinations/persistiq/data/persistIqIdentifyConfig.json:4]: for update mainly id is used , and email cannot be used

**Review by yashasvibajpai — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**